singularity-eos/eos/singularity_eos.cpp

```diff
@@ -343,327 +351,512 @@ int get_sg_eos( // sizing information
-        PTESolverRhoT<EOSAccessor_, Real *, Real **> method(
-            npte, eos_inx, vfrac_tot, sie_tot, &rho_pte(tid, 0), &vfrac_pte(tid, 0),
-            &sie_pte(tid, 0), &temp_pte(tid, 0), &press_pte(tid, 0), lambda_map,
-            &solver_scratch(tid, 0));
-        const bool res_{PTESolver(method)};
-        // const bool res_{pte_closure_josh_offset(
-        //    npte, eos_v.data(), vfrac_tot, sie_tot, &pte_mats(tid, 0), &rho_pte(tid, 0),
-        //    &vfrac_pte(tid, 0), &sie_pte(tid, 0), &temp_pte(tid, 0), &press_pte(tid, 0),
-        //    lambda_map)};
-        // assign local values to global
-        press_v(i) = p_is_inp ? press_v(i) : 0.0;
-        sie_v(i) = s_is_inp ? sie_v(i) : 0.0;
-        temp_v(i) = t_is_inp ? temp_v(i) : 0.0;
-        bmod_v(i) = 0.0;
-        cv_v(i) = 0.0;
-        dpde_v(i) = 0.0;
-        Real vol_sum = 0.0;
-        for (int m = 0; m < npte; ++m) {
-          int mid = pte_mats(tid, m);
-          int midx = pte_idxs(tid, m);
-          Real rho = mass_pte(tid, m) / vfrac_pte(tid, m);
-          Real spie = sie_pte(tid, m);
-          Real press_mat = press_v(i);
-          Real T = temp_v(i) * ev2k;
-          Real cv_mat = 0.0;
-          Real bmod_mat = 0.0;
-          eos_v(mid).FillEos(rho, T, spie, press_mat, cv_mat, bmod_mat, output, cache);
-          frac_sie_v(i, midx) = spie * frac_mass_v(i, midx);
-          frac_vol_v(i, midx) = frac_mass_v(i, midx) / rho;
-          vol_sum += frac_vol_v(i, midx);
-          temp_v(i) += t_is_inp ? 0.0 : frac_vol_v(i, midx) / ev2k * T;
-          press_v(i) += p_is_inp ? 0.0 : frac_vol_v(i, midx) * press_mat;
-          // optional cv and bmod
-          const Real dpde_mat =
-              rho * eos_v(mid).GruneisenParamFromDensityInternalEnergy(rho, spie, cache);
-          sie_v(i) += s_is_inp ? 0.0 : frac_sie_v(i, midx);
-          bmod_v(i) += frac_vol_v(i, midx) * bmod_mat;
-          cv_v(i) += frac_mass_v(i, midx) * cv_mat;
-          dpde_v(i) += frac_vol_v(i, midx) * dpde_mat;
-          if (do_frac_bmod) {
-            frac_bmod_v(i, midx) = bmod_mat;
-          }
-          if (do_frac_cv) {
-            frac_cv_v(i, midx) = cv_mat;
-          }
-          if (do_frac_dpde) {
-            frac_dpde_v(i, midx) = dpde_mat;
-          }
-        }
-        press_v(i) /= p_is_inp ? 1.0 : vol_sum;
-        sie_v(i) /= s_is_inp ? 1.0 : mass_sum;
-        temp_v(i) /= t_is_inp ? 1.0 : vol_sum;
-        bmod_v(i) /= vol_sum;
-        cv_v(i) *= ev2k / mass_sum;
-        dpde_v(i) /= vol_sum;
-        spvol_v(i) = r_is_inp ? spvol_v(i) : vol_sum / mass_sum;
-        pmax_v(i) = press_v(i) > pmax_v(i) ? press_v(i) : pmax_v(i);
-        // if failed, increment return value by 1
-        // use if rather than ternary to reduce contention
-        if (!res_) {
-          res() += 1;
-        }
-        // release token
-        tokens.release(token);
-      });
+  switch(input_int) {
+  case -3:
```

Comment:
Why is this a hardcoded number? Can we make an enum?

Reply:
this is all slated to leave singularity-eos at some point or other and is very EAP specific. It can be enum-ized, but I'm not sure what it buys you.

Reply:
It buys clarity and costs one line of code. :P

Reply:
done